### PR TITLE
Rename Enumerable to Stream

### LINKED ISSUE #25: `enumerable` is a bad name
An `enumerable` is an infinite "stream" of data. Is it the wrong name?

Some alternatives:

* `Stream` - the perfect name IMO, but problematic because it collides with a Kinesis `Stream`.
  * Maybe `CloudStream`?
  * Maybe rename the Kinesis `Stream` to `KinesisStream`?
* `EventSource` - enumerables are built on AWS's event sources, so this kinda makes sense, but does it make sense for chains (map, flatMap, filter, etc.)? 
* `Traversable` - borrowed from scala's library. It sounds wrong because we're not traversing anything - we're responding to new data in an endless stream.
* `Iterable` - does it collide with node.js standard library? Similarly to Traversable, the name doesn't perfectly represent a stream.

I'm leaning towards renaming `Stream` to `KinesisStream` and renaming `Enumerable` to `Stream`, but we then run into some consistency issues. We call SQS Queues `Queue`, and SNS Topics `Topic` - we don't include the service name. Is it weird to do that for Kinesis as an edge case? Based on feedback, we may also want to rename the Glue `Table` to `GlueTable` to avoid collisions with a DynamoDB `Table`, so those two issues align well.

Another option is to use namespaces:
* `Enumerable` => `Stream`
* `Kinesis.Stream`
* `SQS.Queue`
* `Glue.Table`
* `DynamoDB.Table`

I like it because makes everything consistent, but is it a bit noisy?

### PR DESCRIPTION
**Pull request by sam-goodwin**
Fixes #25 

* [x] Rename `Enumerable` to `Stream`
* [x] Move services into their own namespace:
  * [x] Glue
  * [x] DynamoDB
  * [x] SNS
  * [x] SQS
  * [x] Kinesis
  * [x] S3
  * [x] Lambda
* [x] Update Examples
* [x] Update README

### CODE REVIEW

File: examples/lib/stream-processing.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ const stream = queue.enumerable() // enumerable gives us a nice chainable API fo
       };
     }
   })
-  .toStream(stack, 'Stream', {
+  .toKinesis(stack, 'Stream', {
```
Comment (sam-goodwin) [inline examples/lib/stream-processing.ts]: Any thoughts on `toKinesis` vs `toKinesisStream`? I.e. should I include the resource type as a suffix?

* `toSQS` or `toSQSQueue`
* `toSNS` or `toSNSTopic`
* `toKinesis` or `toKinesisStream`
* `toS3DeliveryStream` or `toFirehose`
* `toGlue` or `toGlueTable` (note: it's already `toGlueTable`, so we need to change it for consistency reasons)

**Comment by sam-goodwin** [comment]
While we're namespacing all the things, should `Function` => `Lambda.Function`?

File: packages/punchcard/lib/storage/index.ts [hunk 2]
```diff
@@ -1,4 +1,10 @@
 export * from './codec';
-export * from './dynamodb';
+export * from './compression';
 export * from './glue';
 export * from './s3';
+
+import * as DynamoDB from './dynamodb';
+
+export {
+  DynamoDB
```
Comment (sam-goodwin) [inline packages/punchcard/lib/storage/index.ts]: This was a neat trick!

**Comment by pocesar** [comment]
> 
> 
> While we're namespacing all the things, should `Function` => `Lambda.Function`?

yeah, Function is a global constructor for function, would be good to go ahead and namespace that as well, and make it obvious it's a Lambda Function (and even sounds cooler haha)

File: examples/lib/stream-processing.ts (hunk 1 above)
Comment (pocesar) [reply examples/lib/stream-processing.ts]: it makes sense just to be a little more verbose to where it's going to, LGTM. 

`to{Service}{Resource}` is a good convention, makes it easier to reason. thinking about those services such as SageMaker that have a lot of underlying resources (although couldn't visualize a `toSageMaker{Resource}` from the top of my head)

Comment (sam-goodwin) [reply examples/lib/stream-processing.ts]: Yeah, I think this is a good convention. Updated to reflect that.

I'm not that happy with `toS3DeliveryStream`. Could have gone with `toFirehoseDeliveryStream`, but how to differentiate between destinations (Redshift, ElasticSearch, Splunk, etc.) Created https://github.com/sam-goodwin/punchcard/issues/31 to follow up.

**Pull request merged**